From 00eaee45e4a65242db22b9514872fac60764f21a Mon Sep 17 00:00:00 2001
From: Siddharth Manthan <siddharth_manthan@outlook.com>
Date: Thu, 6 May 2021 23:41:53 +0530
Subject: [PATCH 16/24] arm64: dts: qcom: msm8916-samsung-gt5-common: add
 sensors

This device has ST LIS2HH12 accelerometer and Capella CM3323 light sensor.
Add support for it.

Signed-off-by: Siddharth Manthan <siddharth_manthan@outlook.com>
---
 .../dts/qcom/msm8916-samsung-gt5-common.dtsi  | 37 +++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/msm8916-samsung-gt5-common.dtsi b/arch/arm64/boot/dts/qcom/msm8916-samsung-gt5-common.dtsi
index afc5c4c570..beb3920a83 100644
--- a/arch/arm64/boot/dts/qcom/msm8916-samsung-gt5-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/msm8916-samsung-gt5-common.dtsi
@@ -124,6 +124,35 @@ usb_vbus: usb-vbus {
 	};
 };
 
+&blsp_i2c2 {
+	status = "okay";
+
+	accelerometer@1d {
+		compatible = "st,lis2hh12";
+		reg = <0x1d>;
+
+		vdd-supply = <&pm8916_l17>;
+		vddio-supply = <&pm8916_l5>;
+
+		interrupt-parent = <&msmgpio>;
+		interrupts = <115 IRQ_TYPE_EDGE_RISING>;
+		interrupt-names = "INT1";
+
+		st,drdy-int-pin = <1>;
+		mount-matrix = "0", "1", "0",
+			     "-1", "0", "0",
+			     "0", "0", "1";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&accel_int_default>;
+	};
+
+	light-sensor@10 {
+		compatible = "capella,cm3323";
+		reg = <0x10>;
+	};
+};
+
 &blsp1_uart2 {
 	status = "okay";
 };
@@ -336,6 +365,14 @@ l18 {
 };
 
 &msmgpio {
+	accel_int_default: accel-int-default {
+		pins = "gpio115";
+		function = "gpio";
+
+		drive-strength = <2>;
+		bias-disable;
+	};
+
 	fg_int_default: fg-int-default {
 		pins = "gpio121";
 		function = "gpio";
-- 
2.25.1

